### PR TITLE
Updates for buoy_msgs reorg

### PR DESCRIPTION
**Pull request by andermi**
updates from https://github.com/osrf/buoy_msgs/pull/23

### CODE REVIEW

File: buoy_gazebo/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
-find_package(buoy_msgs REQUIRED)
+find_package(buoy_interfaces REQUIRED)
```
Comment (quarkytale) [inline buoy_gazebo/CMakeLists.txt:20]: nit reorder

File: buoy_tests/package.xml [hunk 2]
```diff
@@ -14,8 +14,9 @@
   <buildtool_depend>rosidl_default_generators</buildtool_depend>
 
   <test_depend>buoy_gazebo</test_depend>
-  <test_depend>buoy_msgs</test_depend>
-  <test_depend>buoy_examples</test_depend>
+  <test_depend>buoy_interfaces</test_depend>
+  <test_depend>buoy_api_cpp</test_depend>
+  <test_depend>buoy_api_py</test_depend>
   <test_depend>rclpy</test_depend>
```
Comment (quarkytale) [inline buoy_tests/package.xml:20]: ditto

File: buoy_tests/CMakeLists.txt [hunk 3]
```diff
@@ -67,7 +67,8 @@ if(BUILD_TESTING)
 
   # Helper function to generate pytest
   function(buoy_add_pytest PYTEST_NAME #[[optional: GTEST_NAME]])
-    find_package(buoy_msgs REQUIRED)
+    find_package(buoy_interfaces REQUIRED)
+    find_package(buoy_api_cpp REQUIRED)
     find_package(rclcpp REQUIRED)
```
Comment (quarkytale) [inline buoy_tests/CMakeLists.txt:72]: nit reorder

**Review by quarkytale — APPROVED** [review]

**Comment by quarkytale** [comment]
This is weird, tests were passing on my local system.

**Comment by andermi** [comment]
> tests were passing

I have no idea what's wrong here... This is from CI logs

```
2022-08-22T23:04:43.1667538Z Starting >>> buoy_msgs
2022-08-22T23:04:43.1696858Z Traceback (most recent call last):
2022-08-22T23:04:43.1697572Z   File "/usr/lib/python3/dist-packages/colcon_core/executor/__init__.py", line 91, in __call__
2022-08-22T23:04:43.1697897Z     rc = await self.task(*args, **kwargs)
2022-08-22T23:04:43.1698340Z   File "/usr/lib/python3/dist-packages/colcon_core/task/__init__.py", line 93, in __call__
2022-08-22T23:04:43.1698674Z     return await task_method(*args, **kwargs)
2022-08-22T23:04:43.1699134Z   File "/usr/lib/python3/dist-packages/colcon_ros/task/ament_cmake/test.py", line 29, in test
2022-08-22T23:04:43.1699662Z     return await extension.test()
2022-08-22T23:04:43.1700100Z   File "/usr/lib/python3/dist-packages/colcon_cmake/task/cmake/test.py", line 45, in test
2022-08-22T23:04:43.1700441Z     assert os.path.exists(args.build_base), \
2022-08-22T23:04:43.1700730Z AssertionError: Has this package been built before?
2022-08-22T23:04:43.1708654Z --- stderr: buoy_msgs
2022-08-22T23:04:43.1708925Z Traceback (most recent call last):
2022-08-22T23:04:43.1709382Z   File "/usr/lib/python3/dist-packages/colcon_core/executor/__init__.py", line 91, in __call__
2022-08-22T23:04:43.1709708Z     rc = await self.task(*args, **kwargs)
2022-08-22T23:04:43.1710136Z   File "/usr/lib/python3/dist-packages/colcon_core/task/__init__.py", line 93, in __call__
2022-08-22T23:04:43.1710467Z     return await task_method(*args, **kwargs)
2022-08-22T23:04:43.1710894Z   File "/usr/lib/python3/dist-packages/colcon_ros/task/ament_cmake/test.py", line 29, in test
2022-08-22T23:04:43.1711228Z     return await extension.test()
2022-08-22T23:04:43.1711655Z   File "/usr/lib/python3/dist-packages/colcon_cmake/task/cmake/test.py", line 45, in test
2022-08-22T23:04:43.1711982Z     assert os.path.exists(args.build_base), \
2022-08-22T23:04:43.1712280Z AssertionError: Has this package been built before?
2022-08-22T23:04:43.1712545Z ---
2022-08-22T23:04:43.1712758Z Failed   <<< buoy_msgs [0.00s, exited with code 1]
```

**Pull request merged**